cloudamqp/resource_cloudamqp_extra_disk_size.go

```diff
@@ -28,27 +26,103 @@ func resourceExtraDiskSize() *schema.Resource {
 				Required:    true,
 				Description: "Extra disk size in GB",
 			},
+			"allow_downtime": {
+				Type:        schema.TypeBool,
+				ForceNew:    true,
+				Required:    true,
+				Description: "When resizing disk, allow downtime to do so",
+			},
+			"nodes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"disk_size": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Subscription plan disk size",
+						},
+						"additional_disk_size": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "Additional added disk size",
+						},
+					},
+				},
+			},
 		},
 	}
 }
 
 func resourceExtraDiskSizeUpdate(d *schema.ResourceData, meta interface{}) error {
-	api := meta.(*api.API)
-	params := make(map[string]interface{})
+	var (
+		api     = meta.(*api.API)
+		params  = make(map[string]interface{})
+		sleep   = 30   // 30 seconds between retires when polling if extra disk size been succesfull.
```

Comment:
> 30 seconds between retires when polling if extra disk size been succesfull

Not sure this is true. The extra disk API is async, so it will return 200 OK when the resize/swap disk action has started. Then [`waitUntilAllNodesReady`](https://github.com/84codes/go-api/blob/ec1089d532a07320b2205ad57892e55349db4ff5/api/instance.go#L33-L57) will run. Not sure if that will actually detect if the extra disk operation has completed?

Reply:
So this value (30 seconds) just controls the time to wait to retry an unsuccessful "add extra disk" operation, right?

Reply:
WaitUntilAllNodesReady will wait for configured to be set to true once the resize disk action is completed. You are correct, needs to be reworked if we want the sleep/timeout to also include this.